pkg/tfvars/libvirt/libvirt.go

```diff
@@ -34,21 +39,44 @@ func TFVars(masterConfig *v1beta1.LibvirtMachineProviderConfig, osImage string,
 		return nil, err
 	}
 
-	osImage, err = cachedImage(osImage)
-	if err != nil {
-		return nil, errors.Wrap(err, "failed to use cached libvirt image")
-	}
-
 	cfg := &config{
 		URI:          masterConfig.URI,
-		Image:        osImage,
 		IfName:       bridge,
 		BootstrapIP:  bootstrapIP.String(),
 		MasterIPs:    masterIPs,
 		MasterMemory: strconv.Itoa(masterConfig.DomainMemory),
 		MasterVcpu:   strconv.Itoa(masterConfig.DomainVcpu),
 	}
 
+	baseImageURL, err := url.ParseRequestURI(osImage)
+	osImage, err = cachedImage(osImage)
+	if strings.HasSuffix(baseImageURL.Path, ".gz") {
+		osImageUncompressed := osImage + ".uncompressed"
+		// Do nothing if we already have the uncompressed file in cache, otherwise decompress the data
+		_, err = os.Stat(strings.TrimPrefix(osImageUncompressed, "file://"))
+		if err != nil {
+			if os.IsNotExist(err) {
+				logrus.Infof("Decompress image data from %v to %v", osImage, osImageUncompressed)
+				err = cache.DecompressFile(strings.TrimPrefix(osImage, "file://"), strings.TrimPrefix(osImageUncompressed, "file://"))
+				if err != nil {
+					logrus.Infof("Decompress %v %v failed: %v", osImage, osImageUncompressed, err)
+					return nil, err
+				}
+			} else {
+				logrus.Infof("Misc failure looking for %v: %v", osImageUncompressed, err)
+				return nil, err
```

Comment:
I guess both of the `logrus.Infof` logs in the error path should be `logrus.Errorf` ?

Reply:
Probably.  Those were actually debugging messages for me, but failures here really should be noted.